handler/handler.go

```diff
@@ -301,6 +301,9 @@ func (h *Handler) prepareUPPAnnotations(ctx context.Context, contentUUID string,
 
 	ann, err := h.annotationsAPI.GetAllButV2(ctx, contentUUID)
 	if err != nil {
+		if err.Error() == annotations.UPPNotFoundMsg {
+			return nil, http.StatusNotFound, err
+		}
```

Comment:
A better solution would be to propagate all errors from upp and not just the `not found`
```suggestion
		var uppErr *annotations.UPPError
		if errors.As(err, &uppErr) {
			return nil,uppErr.Status(), err
		}
```

Reply:
I'm not sure that the general approach is better.
For example, we don't want to return "user error 400" if UPP returns 400, it's not the user's error but an internal error.

Reply:
Fair. Still lets use `errors.As` and then check if it's not found error. Because `err` returned by `annotationsAPI` could be wrapped. And lets compare the `uppErr.Status() == http.StatusNotFound` instead of the message value. This sounds cleaner to me.

Reply:
I also wanted to check Status() but unfortunately errors.As(err, &uppErr) returns false.